third_party/spdk_defs/nvme_defs.h

```diff
@@ -0,0 +1,210 @@
+#ifndef LIB_NVME_DEFS_H
+#define LIB_NVME_DEFS_H
+
+#include <cstdint>
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
If we want to build into a kernel module it is good to have this logic; however, master does not currently have it. Should we add it there?

Reply:
Yeah i think we should add it to scsi_defs as well

Reply:
would recommend having a separate PR for making it a Kernel module as it will be easier to roll back in the future in case we decide to not build this as a kernel module

Reply:
good idea! done